server/fleet/software_installer.go

```diff
@@ -32,3 +31,66 @@ func (FailingSoftwareInstallerStore) Put(ctx context.Context, installerID string
 func (FailingSoftwareInstallerStore) Exists(ctx context.Context, installerID string) (bool, error) {
 	return false, errors.New("software installer store not properly configured")
 }
+
+// SoftwareInstaller represents a software installer package that can be used to install software on
+// hosts in Fleet.
+type SoftwareInstaller struct {
+	// TeamID is the ID of the team. A value of nil means it is scoped to hosts that are assigned to
+	// no team.
+	TeamID *uint `json:"team_id" db:"team_id"`
+	// Name is the name of the software package.
+	Name string `json:"name" db:"name"`
+	// Version is the version of the software package.
+	Version string `json:"version" db:"version"`
+	// UploadedAt is the time the software package was uploaded.
+	UploadedAt string `json:"uploaded_at" db:"uploaded_at"`
+	// InstallerID is the unique identifier for the software package.
+	InstallerID string `json:"-" db:"installer_id"`
+	// InstallScript is the script to run to install the software package.
+	InstallScript string `json:"install_script" db:"install_script"`
+	// PreInstallQuery is the query to run as a condition to installing the software package.
+	PreInstallQuery string `json:"pre_install_query" db:"pre_install_query"`
+	// PostInstallScript is the script to run after installing the software package.
+	PostInstallScript string `json:"post_install_script"`
+}
+
+// AuthzType implements authz.AuthzTyper.
+func (s *SoftwareInstaller) AuthzType() string {
+	return "software_installer"
+}
+
+// SoftwareInstallerStatus represents the status of a software installer package.
+type SoftwareInstallerStatus struct {
```

Comment:
Intended to track the draft API [response](https://github.com/fleetdm/fleet/pull/17865/files#diff-7246bc304b15c8865ed8eaa205e9c244d0a0314e4bae60cf553dc06147c38b64R8205-R8209). Do we want to embed this type in the `SoftwareInstaller` struct?